### ISSUE TITLE
Calico-node somehow causing conntrack to drop tracking of connections

### ISSUE THREAD

**Issue by bradbehle** [projectcalico/calico#1213]
While running a stress test that repeatedly sends a 10MB file using curl to a NodePort service (backed by a few pods), a few times a minutes (so maybe 1 out of 50 or so file transfers) the connection gets reset so the file transfer fails.  Note that this cluster has 165 worker nodes (but this test only runs two server pods)

## Expected Behavior
These files should all be able to transfer successfully.

## Current Behavior
Some of these transfers fail.  I have tcpdump traces from the client pod cali* interface, client tunl0 interface, client bond0 interface, server bond0 interface, and server tunl0 interface.  From these traces I was able to eventually figure out that the connections that were failing were doing so because the linux kernel was sending TCP RST packets to the client sometimes, and to the server sometimes as well.

The best we can tell from the traces, these TCP RST packets are being sent by the linux kernel because for some reason, in the middle of some tcp connections (where the test client is sending a 10MB file via HTTP-POST), conntrack somehow stops tracking it. So a connection that has been made to one of the pods, packets are flowing from the client to that pod, and then connection tracking for that connection stops, and sometimes the next packets (via the random function in iptables that handles kube services) get sent to a different pod (causing that pod to send back RST packets because it doesn't expect or understand the packets), and/or the ack packets from the pod to the client get dropped and RST packets sent in return because linux doesn't know what to do with the return packets due to conntrack disappearing for this connection.

I haven't been able to determine exactly why this connection tracking is being dropped, but it does appear to be due to something that calico-node is doing. When I run the stress test I consistently get several errors a minute. However, if I stop the calico-node container (`sudo systemctl stop calico-node`) on the worker node that the test is connecting to via `workerIP:<NODE_PORT>` (the worker node where the conntrack is being dropped), then I can run the test continuously for about 2 minutes without any errors.  Of course this only works until the return routes go away on the nodes running the server pods, making all calls fail until calico-node is started again.

I have captured the calico and syslog logs from the node where conntrack connections are dropping, I will attach them here

## Steps to Reproduce (for bugs)
I am not sure how easy this is to reproduce outside of our cluster, but here are the steps we used.  I'm guessing any pod that accepts any kind of file (large enough that the transfer takes a second or two) will work for this, I don't think it is specific to using HTTP-POST)
1. Start a deployment with several pods (spread across several worker nodes) that can accept a file via HTTP-POST
2. Start a client pod, create a 10MB file of some kind, and use the following commands to send the file to the server pod:
```
doit() {  curl -k "http://<WORKER_IP>:<NODEPORT>/fileTransfer" -F "images_file=@10MBFile.zip";}
export -f doit

#Test that it works
doit

#Run the test
seq 1000 | parallel -j20 doit
```
3. you can run the above test over and over, and can increase the seq value to make more calls, or reduce it to make less

## Your Environment
* Calico version 2.1.5 (calico-node running in systemd)
* Orchestrator version: kubernetes 1.5.6
* 165 worker nodes, full BGP mesh
* Operating System and version: Ubuntu 16.04.2 x86_64
* Link to your project (optional):

**Comment by bradbehle** [comment]
[calico_logs.zip](https://github.com/projectcalico/calico/files/1397190/calico_logs.zip)
[kern.log.zip](https://github.com/projectcalico/calico/files/1397191/kern.log.zip)
[syslog.zip](https://github.com/projectcalico/calico/files/1397192/syslog.zip)

**Comment by fasaxc** [comment]
@bradbehle Are you able to try this test with Calico v2.6?  We have made fixes in BIRD and Felix since v2.1 that may help (namely to prevent route flaps in various situations).

Failing that, it'd be useful to check `ip monitor` during your test to see if the routes are being removed and re-added and `conntrack -E` to see whether any conntrack deletions occur for the entries that should be causing the NAT to occur.

**Comment by bradbehle** [comment]
@fasaxc We did just try this with Calico 2.6.1 and we can't reproduce it, so it does seem like something might have been fixed since Calico 2.1.5 that is fixing this issue.  So would removing and re-adding the same route (I assume that is what you mean by route flapping) cause the conntrack deletions of connections in progress?

**Comment by fasaxc** [comment]
@bradbehle I'm not sure, but a couple of possibilities:

- some interaction between the RPF check and conntrack
- BIRD removing an IPIP route so packet takes the wrong path and doesn't match.

If you're able to repro with the older code and watch `ip monitor` and `conntrack -E`, that should give a hint as to what route was flapped (or lead us in a different direction).

**Comment by makhov** [comment]
Maybe this post is related to the issue:
https://tech.xing.com/a-reason-for-unexplained-connection-timeouts-on-kubernetes-docker-abd041cf7e02

**Comment by caseydavenport** [comment]
> Calico version 2.1.5 (calico-node running in systemd)

> We did just try this with Calico 2.6.1 and we can't reproduce it, 

I'm going to close this as fixed in a modern release since v2.1 is quite old now.
